c++/src/kj/compat/tls.c++

```diff
@@ -160,11 +161,21 @@ public:
       }
     });
   }
+
   kj::Promise<void> accept() {
     // We are the server. Set SSL options to prefer server's cipher choice.
     SSL_set_options(ssl, SSL_OP_CIPHER_SERVER_PREFERENCE);
 
-    return sslCall([this]() { return SSL_accept(ssl); }).ignoreResult();
+    return sslCall([this]() {
+             return SSL_accept(ssl);
+           })
+        .then([](size_t ret) {
+          if (ret == 0) {
+            kj::throwRecoverableException(
+                KJ_EXCEPTION(DISCONNECTED, "Client disconnected during SSL_accept()"));
+          }
+          KJ_ASSERT(ret > 0, "sslCall must return a non-negative value");
+        });
```

Comment:
This one is still out of style, maybe:
```c++
return sslCall([this]() {
  return SSL_accept(ssl);
}).then([](size_t ret) {
  if (ret == 0) {
    kj::throwRecoverableException(
        KJ_EXCEPTION(DISCONNECTED, "Client disconnected during SSL_accept()"));
  }
  KJ_ASSERT(ret > 0, "sslCall must return a non-negative value");
});
```

or

```c++
return sslCall([this]() { return SSL_accept(ssl); })
    .then([](size_t ret) {
  if (ret == 0) {
    kj::throwRecoverableException(
        KJ_EXCEPTION(DISCONNECTED, "Client disconnected during SSL_accept()"));
  }
  KJ_ASSERT(ret > 0, "sslCall must return a non-negative value");
});
```
You could even pull the first `.then` up, AFAIK.

Reply:
Thanks for pointing this out. Sadly, I can't because of how sslCall works.

Reply:
> I can't because of how sslCall works.

Can't what?

Reply:
The consolidation of the two `then()` lambdas can't happen because sslCall returns a promise, not a size_t.

Reply:
Hmm... my snippets are only style/indentation changes from the code I see above in the diff view.

Reply:
Yeah I only see one `.then()` here... are we looking at the same code?

Reply:
I think we are and I'm bad at explaining myself. We have two lambdas. I'd interpreted @bretthoerner's comment as saying that I could use the output of SSL_accept() in the same lambda to check for disconnect. I don't think we can do that because sslCall can be async. I might have misread the entire convo of course.

Reply:
Oh, no, my whole comment was style related. I meant "pull it up" as in remove a newline. But it looks like this code has changed a bunch since then anyway so this is moot.